adb/filesync_protocol.py

```diff
@@ -161,7 +161,7 @@ def Send(self, command_id, data='', size=0):
       self._Flush()
 
     header = struct.pack('<2I', self.id_to_wire[command_id], size)
-    self.send_buffer += header + data
+    self.send_buffer += header.decode('ascii') + data
```

Comment:
Does this work for all sizes? If size = 128, then a byte in header will be 0x80, which is not in 7-bit ASCII's range. 

```py
In [3]: struct.pack('<I', 128).decode('ascii')
---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)
<ipython-input-3-b48ec2057ace> in <module>()
----> 1 struct.pack('<I', 128).decode('ascii')

UnicodeDecodeError: 'ascii' codec can't decode byte 0x80 in position 0: ordinal not in range(128)
```

It looks like you're converting the header to unicode since data might be unicode, but I think you need to convert data to be bytes using `encode('utf8')` so that we only send bytes over the wire.